tests/test_configs.py

```diff
@@ -90,3 +91,95 @@ def test_o1_requires_temp_equals_1() -> None:
         warnings.simplefilter("always")
         _ = Settings(llm="o1-thismodeldoesnotexist", temperature=1)
         assert not w
+
+
+@pytest.mark.parametrize(
+    "test_case",
+    [
+        pytest.param(
+            {
+                "doc_class": Doc,
+                "doc_data": {
+                    "docname": "Test Paper",
+                    "citation": "Test Citation",
+                    "dockey": "key1",
+                },
+                "filter_criteria": {"docname": "Test Paper"},
+                "expected_result": True,
+            },
+            id="Doc-matching-docname",
+        ),
+        pytest.param(
+            {
+                "doc_class": Doc,
+                "doc_data": {
+                    "docname": "Test Paper",
+                    "citation": "Test Citation",
+                    "dockey": "key1",
+                },
+                "filter_criteria": {"docname": "Another Paper"},
+                "expected_result": False,
+            },
+            id="Doc-nonmatching-docname",
+        ),
+        pytest.param(
+            {
+                "doc_class": DocDetails,
+                "doc_data": {
+                    "title": "Test Paper",
+                    "authors": ["Alice", "Bob"],
+                    "year": 2020,
+                },
+                "filter_criteria": {"title": "Test Paper"},
+                "expected_result": True,
+            },
+            id="DocDetails-matching-title",
+        ),
+        pytest.param(
+            {
+                "doc_class": DocDetails,
+                "doc_data": {
+                    "title": "Test Paper",
+                    "authors": ["Alice", "Bob"],
+                    "year": 2020,
+                },
+                "filter_criteria": {"!year": 2020, "?foo": "bar"},
+                "expected_result": False,
+            },
+            id="DocDetails-inverted-matching-year",
+        ),
+        pytest.param(
+            {
+                "doc_class": DocDetails,
+                "doc_data": {
+                    "title": "Test Paper",
+                    "authors": ["Alice", "Bob"],
+                    "year": 2020,
+                },
+                "filter_criteria": {"year": 2020, "foo": "bar"},
+                "expected_result": False,
+            },
+            id="DocDetails-missing-param-fail",
+        ),
+        pytest.param(
+            {
+                "doc_class": DocDetails,
+                "doc_data": {
+                    "title": "Test Paper",
+                    "authors": ["Alice", "Bob"],
+                    "year": 2020,
+                },
+                "filter_criteria": {"?volume": "10", "!title": "Another Paper"},
+                "expected_result": True,
+            },
+            id="DocDetails-relaxed-missing-volume",
+        ),
+    ],
+)
+def test_matches_filter_criteria(test_case):
```

Comment:
Why not have `doc_class` and others be a separate field outside of the `dict`?

So the first argument to `pytest.mark.parameterize` would be `("test_class", "doc_data", ""filter_criteria", "expected_result")`, and then:

```python
def test_matches_filter_criteria(test_class, doc_data, filter_criteria, expected_result) -> None:
```